snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,152 @@
+        return FunctionCallMatch(
+            String("replaceAll"),
+            (
+                FunctionCallMatch(
+                    String("toString"),
+                    (
+                        Param(
+                            "formatted_uuid_column" + suffix,
+                            ColumnMatch(None, self.__uuid_column_match),
+                        ),
+                    ),
+                ),
+            ),
+            with_optionals=True,
+        )
+
+    def uuid_column_pattern(self, suffix: str = "") -> Param[Column]:
+        return Param(
+            "uuid_column" + suffix, ColumnMatch(None, self.__uuid_column_match)
+        )
+
+    def __init__(self, uuid_columns: Set[str]) -> None:
+        self.__unique_uuid_columns = set(uuid_columns)
+        self.__uuid_column_match = Or(
+            [String(u_col) for u_col in self.__unique_uuid_columns]
+        )
+        self.uuid_in_condition = FunctionCallMatch(
+            Or((String(ConditionFunctions.IN), String(ConditionFunctions.NOT_IN))),
+            (
+                Or((self.uuid_column_pattern(), self.formatted_uuid_pattern())),
+                Param("params", FunctionCallMatch(String("tuple"), None)),
+            ),
+        )
+        self.uuid_condition = FunctionCallMatch(
+            Or([String(op) for op in FUNCTION_TO_OPERATOR]),
+            (
+                Or(
+                    (
+                        Param("literal_0", LiteralMatch(AnyOptionalString())),
+                        self.uuid_column_pattern("_0"),
+                        self.formatted_uuid_pattern("_0"),
+                    )
+                ),
+                Or(
+                    (
+                        Param("literal_1", LiteralMatch(AnyOptionalString())),
+                        self.uuid_column_pattern("_1"),
+                        self.formatted_uuid_pattern("_1"),
+                    )
+                ),
+            ),
+        )
+
+    def parse_uuid(self, lit: Expression) -> Expression:
+        if not isinstance(lit, Literal):
+            return lit
+
+        try:
+            parsed = uuid.UUID(str(lit.value))
+            return Literal(lit.alias, str(parsed))
+        except Exception:
+            return lit
+
+    def process_condition(self, exp: Expression) -> Expression:
+        if not isinstance(exp, FunctionCall):
+            return exp
+
+        result = self.uuid_in_condition.match(exp)
+        if result is not None:
+            new_params: List[Expression] = []
+            if result.contains("formatted_uuid_column"):
+                column = result.expression("formatted_uuid_column")
+                assert isinstance(column, Column)
+                new_params.append(Column(None, column.table_name, column.column_name))
+            else:
+                column = result.expression("uuid_column")
+                assert isinstance(column, Column)
+                new_params.append(column)
```

Comment:
I don't think we need to deal with this case.
If I get it right this identifies `event_id IN tuple('a7d67cf796774551a95be6543cacd459')` to transform it in `event_id IN tuple('a7d67cf7-9677-4551-a95b-e6543cacd459')`.
That to me seems just an invalid query we should reject it instead of trying to fix it as it is a type error.

I would just skip that case, it would simplify the logic of this processor, which is already quite complex.

Reply:
Why is that an invalid query? It seems like a very possible use case to have IN conditions on event_ids. Why not optimize that case?

Reply:
I am talking about the transformation between `a7d67cf796774551a95be6543cacd459` into `a7d67cf7-9677-4551-a95b-e6543cacd459` when the original condition had not been processed into `replaceAll(toString.....))`. 
If, at this point of the query processing, we receive `event_id IN tuple('a7d67cf796774551a95be6543cacd459')` regarding an event_id column that is of type UUID, that query is invalid (that condition, if it was a valid case, should have been transformed earlier so either we failed in the transformation before or the query was invalid to begin with).

I would not fix an invalid query turning it into a valid one in a processor that is supposed to deal with performance only. Otherwise we silently start depending on this processor to be executed here for the correctness of the queries as it covers either a bug in the event id processor or in the translator. Or invalid queries from the client.

Reply:
Ah, I see your point here. OK I'll change this.